sklearn/metrics/plot/roc_curve.py

```diff
@@ -0,0 +1,191 @@
+    auc_ :
+        Area under ROC curve.
+    name_ : str
+        Name of estimator.
+    line_ : matplotlib Artist
+        ROC Curve.
+    ax_ : matplotlib Axes
+        Axes with ROC curv
+    figure_ : matplotlib Figure
+        Figure containing the curve
+    """
+
+    def __init__(self, estimator, X, y, *, pos_label=None, sample_weight=None,
+                 drop_intermediate=True, response_method="auto"):
+        """Computes and stores values needed for visualization"""
+
+        if response_method != "auto":
+            prediction_method = getattr(estimator, response_method, None)
+            if prediction_method is None:
+                raise ValueError(
+                    "response method {} not defined".format(response_method))
+        else:
+            predict_proba = getattr(estimator, 'predict_proba', None)
+            decision_function = getattr(estimator, 'decision_function', None)
+            prediction_method = predict_proba or decision_function
+
+        if prediction_method is None:
+            if response_method == 'predict_proba':
+                raise ValueError('The estimator has no predict_proba method')
+            else:
+                raise ValueError(
+                    'The estimator has no decision_function method')
+
+        y_pred = prediction_method(X)
+
+        if y_pred.ndim != 1:
+            if y_pred.shape[1] > 2:
+                raise ValueError("Estimator must be a binary classifier")
+            y_pred = y_pred[:, 1]
+        fpr, tpr, _ = roc_curve(y, y_pred, pos_label=pos_label,
+                                drop_intermediate=drop_intermediate)
+
+        self.fpr_ = fpr
+        self.tpr_ = tpr
+        self.auc_ = auc(fpr, tpr)
+        self.name_ = estimator.__class__.__name__
+
+    def plot(self, ax=None, line_kw=None, name=None):
+        """Plot visualization
+
+        Parameters
+        ----------
+        ax : Matplotlib Axes, default=None
+            Axes object to plot on.
+
+        line_kw : dict or None, default=None
+            Keyword arguments to pass to.
+
+        name : str or None, default=None
+            Name of ROC Curve for labeling. If `None`, use the name of the
+            estimator.
+        """
+        check_matplotlib_support('plot_roc_curve')  # noqa
+        import matplotlib.pyplot as plt  # noqa
+
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        if name is None:
+            name = self.name_
+
+        label = "{} (AUC = {:0.2f})".format(name, self.auc_)
+        line_kwargs = {"label": label}
+        if line_kw is not None:
+            line_kwargs.update(**line_kw)
+        self.line_ = ax.plot(self.fpr_, self.tpr_, **line_kwargs)[0]
+        ax.set_xlabel("False Positive Rate")
+        ax.set_ylabel("True Positive Rate")
+        ax.legend()
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_roc_curve(estimator, X, y, pos_label=None, sample_weight=None,
```

Comment:
If there's only one artist, I think we don't need to have a dict as an argument and can just do ``**kwargs`` instead of having ``line_kw={...}``